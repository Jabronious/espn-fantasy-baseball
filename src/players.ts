import { BaseClass } from './base-class';
import { ESPNCookiesDto } from './models/classes/espn-cookies.dto';
import { PlayerDto, PlayerResponseDto } from './models/classes/player.dto';

export class Players extends BaseClass {
	constructor(leagueId: number, cookies?: ESPNCookiesDto) {
		super(leagueId, cookies);
	}

	/**
	 * @param {number} playerId - player id for the player you wish to look up
	 * @param {boolean} [detailed=false] - true will returned more detailed data about a player
	 * @returns Player data for the given team id
	 */
	async getPlayerById(playerId: number, detailed: boolean = false): Promise<PlayerDto> {
		// https://fantasy.espn.com/apis/v3/games/flb/seasons/2022/players/32758?view=players_wl
		const params = { view: detailed ? 'mRoster' : 'players_wl' };
		const playerResponse = await this.fantasyRequests
			.get<PlayerResponseDto>(`/players/${playerId}`, {}, params)
			.catch((e) => {
				throw new Error(e);
			});

		return playerResponse.player;
	}
}
