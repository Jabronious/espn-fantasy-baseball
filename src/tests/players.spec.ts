import axios from 'axios';
import { Players } from '../players';

const player = {
	defaultPositionId: 3,
	droppable: true,
	eligibleSlots: [1, 7, 19, 12, 16, 17],
	firstName: 'Christian',
	fullName: 'Christian Walker',
	id: 32758,
	lastName: 'Walker',
	ownership: {
		percentOwned: 4.399557277255119,
	},
	proTeamId: 29,
	universeId: 2,
};
describe('Players', () => {
	const players = new Players(1);

	it('should be defined', () => {
		expect(players).toBeDefined();
	});

	describe('getPlayerById', () => {
		it('should get simple version of the player', async () => {
			jest.spyOn(axios, 'request').mockResolvedValueOnce({ data: { player: { ...player } } });

			expect(await players.getPlayerById(32758)).toMatchSnapshot();
		});

		it('should catch errors and throw them', async () => {
			jest.spyOn(axios, 'request').mockRejectedValueOnce('error');

			await expect(players.getPlayerById(32758)).rejects.toMatchSnapshot();
		});
	});
});
