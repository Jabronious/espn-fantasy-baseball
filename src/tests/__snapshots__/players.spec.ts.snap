// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players getPlayerById should catch errors and throw them 1`] = `[Error: error]`;

exports[`Players getPlayerById should get simple version of the player 1`] = `
Object {
  "defaultPositionId": 3,
  "droppable": true,
  "eligibleSlots": Array [
    1,
    7,
    19,
    12,
    16,
    17,
  ],
  "firstName": "Christian",
  "fullName": "Christian Walker",
  "id": 32758,
  "lastName": "Walker",
  "ownership": Object {
    "percentOwned": 4.399557277255119,
  },
  "proTeamId": 29,
  "universeId": 2,
}
`;
